### PR TITLE
Add chatgpt.rb

### PR DESCRIPTION
**Pull request by Aaron-212**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=pullrequests).
- [x] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [x] `brew audit --cask --new <cask>` worked successfully.
- [x] `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

---

Double checked the url, an OpenAI official says https://persistent.oaistatic.com/sidekick/public/ChatGPT_Desktop_public_latest.dmg is the legit download link.

Mentioned here: https://community.openai.com/t/chat-gpt-desktop-app-for-mac/744613/40

### CODE REVIEW

**Comment by bevanjkay** [comment]
Thanks @Aaron-212 - I think we should wait until there is more information regarding public availability.

File: Casks/c/chatgpt.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  sha256 :no_check
+
+  url "https://persistent.oaistatic.com/sidekick/public/ChatGPT_Desktop_public_latest.dmg",
+      verified: "persistent.oaistatic.com/sidekick/public/"
+  name "ChatGPT"
+  desc "OpenAI official ChatGPT desktop app for GPT-4o / GPT-3.5"
+  homepage "https://chatgpt.com/"
+
+  depends_on macos: ">= :sonoma"
+
```
Comment (p-linnane) [inline Casks/c/chatgpt.rb:12]:
```suggestion
  depends_on macos: ">= :sonoma"
  depends_on arch: :arm64

```

File: Casks/c/chatgpt.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
+cask "chatgpt" do
+  version :latest
+  sha256 :no_check
+
+  url "https://persistent.oaistatic.com/sidekick/public/ChatGPT_Desktop_public_latest.dmg",
+      verified: "persistent.oaistatic.com/sidekick/public/"
+  name "ChatGPT"
+  desc "OpenAI official ChatGPT desktop app for GPT-4o / GPT-3.5"
```
Comment (p-linnane) [inline Casks/c/chatgpt.rb:8]:
```suggestion
  desc "OpenAI's official ChatGPT desktop app"
```

Comment (p-linnane) [reply Casks/c/chatgpt.rb:8]: Let's make this generic so we don't have to update it when a new model is released.

File: Casks/c/chatgpt.rb (hunk 1 above)
Comment (p-linnane) [reply Casks/c/chatgpt.rb:12]:
```
file -b /Volumes/ChatGPT\ Installer/ChatGPT.app/Contents/MacOS/ChatGPT
Mach-O 64-bit executable arm64
```

**Review by p-linnane — APPROVED** [review]
Going to merge this now. Users who don't have access will get an error message when logging in, but that's not a Homebrew problem.

**Pull request merged**